kit/azure/networking/main.tf

```diff
@@ -0,0 +1,54 @@
+data "azurerm_subscription" "current" {
+}
+
+#TODO setting the name doesn't work. Azure says alias is already in use. Importing fails because I can't figure out which alias to use for that.
+# resource "azurerm_subscription" "networking" {
+#   subscription_id   = data.azurerm_subscription.current.subscription_id
+#   subscription_name = "hub"
+# }
+
+resource "azurerm_management_group_subscription_association" "this" {
+  subscription_id = data.azurerm_subscription.current.id
+  management_group_id = var.parent_management_group_id
+}
+
+# Permissions for deploy user
+resource "azurerm_role_definition" "cloudfoundation_tfdeploy" {
+  name  = "hub_networking" #TODO definition names are unique per tenant. make it configurable
+  scope = data.azurerm_subscription.current.id
+  permissions {
+    actions = [
+      "Microsoft.Resources/subscriptions/resourceGroups/write",
+      "Microsoft.Resources/subscriptions/resourceGroups/delete",
+    ]
+  }
+}
+
+resource "azurerm_role_assignment" "cloudfoundation_tfdeploy" {
+  principal_id       = var.cloudfoundation_deploy_principal_id
+  scope              = data.azurerm_subscription.current.id
+  role_definition_id = azurerm_role_definition.cloudfoundation_tfdeploy.role_definition_resource_id
+}
+
+data "azurerm_role_definition" "network_contributor" {
+  name = "Network Contributor"
+}
+
+resource "azurerm_role_assignment" "network_contributor" {
+  principal_id       = var.cloudfoundation_deploy_principal_id
+  scope              = data.azurerm_subscription.current.id
+  role_definition_id = data.azurerm_role_definition.network_contributor.id
```

Comment:
t: azurerm_role_assignment supports a `role_definition_name` nowadays, which shortens the code a bit since you no longer need a data object to look up the role_definition_id